platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Utilities.java

```diff
@@ -380,8 +385,47 @@ public static boolean hasAnyCauseSuppliedType(
                 .map(address -> new PeerInfo(
                         address.getNodeId(),
                         address.getSelfName(),
-                        address.getHostnameExternal(),
-                        address.getSigCert()))
+                        Objects.requireNonNull(address.getHostnameExternal()),
+                        Objects.requireNonNull(address.getSigCert())))
                 .toList();
     }
+
+    /**
+     * Validates that there exists a known TLS peer whose signing certificate's subject
+     *      matches the agreement certificate's issuer
+     *
+     * @param sslsocket
+     * 		the address book to create the list from
+     * @param peerInfoList
+     * 		List of peers to validate
+     * @return info of the corresponding peer
+     * */
+    public static PeerInfo validateTLSPeer(
+            final @NonNull SSLSocket sslsocket, final @NonNull List<PeerInfo> peerInfoList) {
+        PeerInfo matchingPair = null;
+        try {
+            final Certificate[] certs = sslsocket.getSession().getPeerCertificates();
+            final X509Certificate agreementCert = (X509Certificate) certs[0];
+            final Optional<PeerInfo> peerOptional = peerInfoList.stream()
+                    .filter(peerInfo -> ((X509Certificate) peerInfo.signingCertificate())
+                            .getSubjectX500Principal()
+                            .equals(agreementCert.getIssuerX500Principal()))
+                    .findFirst();
+            if (peerOptional.isEmpty()) {
+                logger.warn(
```

Comment:
Should these logs be `warn`? Seems severe enough for `error` IMO

Reply:
someone else connecting to a server socket and getting closed out due to not being a 'valid' connection should not be an error (it doesn't lead to a fault in the app). I think it's ok to warn and move on.